### PR TITLE
MINOR: Use `Map.forKeyValue` to avoid tuple allocation in Scala 2.13

### PR DESCRIPTION
**Pull request by ijuma**
`foreachKv` invokes `foreachEntry` in Scala 2.13 and falls back to
`foreach` in Scala 2.12.

This change requires a newer version of scala-collection-compat, so
update it to the latest version (2.2.0).

Finally, included a minor clean-up in `GetOffsetShell` to use `toArray`
before `sortBy` since it's more efficient.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/tools/GetOffsetShell.scala [hunk 1]
```diff
@@ -133,7 +133,7 @@ object GetOffsetShell {
         }
     }
 
-    partitionOffsets.toSeq.sortBy { case (tp, _) => tp.partition }.foreach { case (tp, offset) =>
+    partitionOffsets.toArray.sortBy { case (tp, _) => tp.partition }.foreach { case (tp, offset) =>
```
Comment (ijuma) [inline core/src/main/scala/kafka/tools/GetOffsetShell.scala:136]: Unrelated clean-up I noticed.

**Review by jsancio — APPROVED** [review]
LGTM! Thanks for the improvement.

File: core/src/main/scala/kafka/utils/Implicits.scala [hunk 2]
```diff
@@ -46,4 +47,21 @@ object Implicits {
...
+   * (with the help of scala.collection.compat). `foreachEntry` avoids the tuple allocation and
+   * is more efficient.
+   *
+   * This was not named `foreachEntry` to avoid `unused import` warnings in Scala 2.13 (the implicit
+   * would not be triggered in Scala 2.13 since `Map.foreachEntry` would have precedence).
+   */
+  @nowarn("cat=unused-imports")
+  implicit class MapExtensionMethods[K, V](private val self: scala.collection.Map[K, V]) extends AnyVal {
+    import scala.collection.compat._
+    def foreachKv[U](f: (K, V) => U): Unit = {
```
Comment (hachikuji) [inline core/src/main/scala/kafka/utils/Implicits.scala]: The name reads a tad awkwardly. I wonder if `foreachKeyValue` would be too verbose. Or maybe `foreachMapEntry`?

**Review by hachikuji — APPROVED** [review]
LGTM. Added a couple alternative names for consideration, but will leave it up to you.

File: core/src/main/scala/kafka/utils/Implicits.scala (hunk 2 above)
Comment (jsancio) [reply core/src/main/scala/kafka/utils/Implicits.scala]: Or `foreachPair`.

Comment (ijuma) [reply core/src/main/scala/kafka/utils/Implicits.scala]: Thanks for the feedback. I agree that the name could be improved. I was hoping for something concise and descriptive. `foreachPair` is not bad, but you'd expect to get a pair (tuple) and this is to avoid that. How about `forKeyValue`?

Comment (ijuma) [reply core/src/main/scala/kafka/utils/Implicits.scala]: Also, note that this will go away once we drop support for Scala 2.12.

Comment (jsancio) [reply core/src/main/scala/kafka/utils/Implicits.scala]: `forKeyValue` LGTM since `for` is the language's keyword for `foreach` or `map`.

Comment (ijuma) [reply core/src/main/scala/kafka/utils/Implicits.scala]: That was the motivation indeed. Great, I'll change it.

**Comment by dajac** [comment]
'forKeyValue' sounds good to me as well.

File: core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala [hunk 3]
```diff
@@ -128,7 +129,7 @@ object ZkSecurityMigrator extends Logging {
     // Now override any set system properties with explicitly-provided values from the config file
     // Emit INFO logs due to camel-case property names encouraging mistakes -- help people see mistakes they make
     info(s"Found ${zkTlsConfigFileProps.size()} ZooKeeper client configuration properties in file $filename")
-    zkTlsConfigFileProps.asScala.foreach { case (key, value) =>
+    zkTlsConfigFileProps.asScala.forKeyValue { (key, value) =>
```
Comment (chia7712) [inline core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala:132]: Could we replace such ```asScala.forKeyValue``` by java ```Map.forEach``` ? We don't need to convert it to scala collection actually.

Comment (ijuma) [reply core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala:132]: See https://github.com/apache/kafka/commit/847ff8f55735568fe879d490af297e047b994f5e for an explanation why `asScala` still makes sense when dealing with `Properties`.

File: core/src/main/scala/kafka/utils/Implicits.scala (hunk 2 above)
Comment (ijuma) [reply core/src/main/scala/kafka/utils/Implicits.scala]: Checked with @hachikuji and he's fine with this change.
